src/Entity/Adherent.php

```diff
@@ -1004,29 +957,38 @@ public function setLegislativeCandidate(bool $candidate): void
         $this->legislativeCandidate = $candidate;
     }
 
-    public function getComMobile(): ?bool
+    public function allowNotification(): ?bool
     {
-        return $this->comMobile;
+        return $this->allowNotification;
     }
 
-    public function setComMobile(?bool $comMobile): void
+    public function addSubscriptionType(SubscriptionType $type): void
     {
-        $this->comMobile = $comMobile;
+        if (!$this->subscriptionTypes->contains($type)) {
+            $this->subscriptionTypes->add($type);
+        }
     }
 
-    public function setComEmail(?bool $comEmail): void
+    public function removeSubscriptionType(SubscriptionType $type): void
     {
-        $this->setCitizenProjectCreationEmailSubscriptionRadius(
-            $comEmail ? self::CITIZEN_PROJECT_EMAIL_DEFAULT_DISTANCE : self::DISABLED_CITIZEN_PROJECT_EMAIL
-        );
+        $this->subscriptionTypes->removeElement($type);
+    }
 
-        $this->localHostEmailsSubscription = $comEmail;
+    public function removeSubscriptionTypeByCode(string $code): void
```

Comment:
When is this method used?

What about simply:
```php
foreach ($this->subscriptionTypes as $type) {
     if ($code === $type->getCode()) {
        $this->removeSubscriptionType($type);
    }
}
```
Not sure we need to introduce Doctrine related complexity to the model.